server/clients/tfgrid/vm_with_gw.go

```diff
@@ -0,0 +1,344 @@
+package tfgrid
+
+import (
+	"context"
+	"fmt"
+	"math/rand"
+	"strings"
+
+	"github.com/rs/zerolog/log"
+	"github.com/threefoldtech/zos/pkg/gridtypes/zos"
+)
+
+const gwNameEnvVar = "WEB3PROXY_DOMAIN_NAME"
+const letters = "abcdefghijklmnopqrstuvwxyz"
```

Comment:
I asked to rename this file to vm.go instead as you can deploy vms without gateway. Please do that for all objects/files/classes that are named like this